### DISCUSSION TITLE
Issues

### DISCUSSION THREAD

**Discussion by MrSteffj** [mriscoc/Ender3V2S1#437] [Q&A] [answered]
Hello,

First and foremost thanks for this awesome firmware.

I have some issues with bed leveling and starting prints from the SD card.

Whenever I start a print, via OctoPrint or the SD card, it will always automatically try to level the bed once the bed is heated. Also if a mesh is available. I can't seem to turn it off.

When I print from the SD card, nearly of the filament gets retracted out of the tubing just right before printing the first line, because of this the print will start without any filament. I can't find a way to turn this off.

Any help would be appreciated
^^

Thanks

**Comment by ghost** [comment, marked as answer]
> Hello,
> 
> First and foremost thanks for this awesome firmware.
> 
> I have some issues with bed leveling and starting prints from the SD card.
> 
> Whenever I start a print, via OctoPrint or the SD card, it will always automatically try to level the bed once the bed is heated. Also if a mesh is available. I can't seem to turn it off.
> 
> When I print from the SD card, nearly of the filament gets retracted out of the tubing just right before printing the first line, because of this the print will start without any filament. I can't find a way to turn this off.
> 
> Any help would be appreciated ^^
> 
> Thanks

Hello, you need to check your gcode, it may be that the slicer has gcode to make leveling for each print and also the filament issue.
If you can post a file of yours that is used to print, I'll check it

**Reply by MrSteffj** [reply]
Thanks for the reply, it was indeed in the gcode.

I did not remember putting it there, so I didn't think of checking it out.

Issue resolved, thanks again :)

**Comment by DanMan32** [comment]
I had the same problem of filament unloading with one of my calibration prints, and not even with this firmware.  Turns out it was having added gcode for the priming line I grabbed from a Cura sliced gcode, but didn't include "G92 E0 ; Reset Extruder" that was executed higher up in the Cura generated gcode.
By default positioning is absolute, so when you power on, you are at extruder position 0.   If you then perform any filament management  such as purging, then the extruder position is offset.
Then when you print, typically the extruder is moved to absolute 0 position, which isn't what the printer now considers the 0 position.  Lets say it now considers the extruder to be at 100mm position.  Your file then tells the extruder to move to position 10mm when it is expecting to print a 10mm line, but in reality the 10mm position is currently 90mm back.
Rule of thumb: be sure you have gcode in your file to reset the extruder to 0 before you do any extruder movement.
